engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/AbstractStoragePoolAllocator.java

```diff
@@ -219,13 +206,21 @@ protected boolean filter(ExcludeList avoid, StoragePool pool, DiskProfile dskCh,
             return false;
         }
 
+        if (pool.isManaged() && !managedStoragePoolCanScale(pool, plan)) {
+            return false;
+        }
+
         // check capacity
-        Volume volume = _volumeDao.findById(dskCh.getVolumeId());
-        List<Volume> requestVolumes = new ArrayList<Volume>();
+        Volume volume = volumeDao.findById(dskCh.getVolumeId());
+        List<Volume> requestVolumes = new ArrayList<>();
         requestVolumes.add(volume);
         return storageMgr.storagePoolHasEnoughIops(requestVolumes, pool) && storageMgr.storagePoolHasEnoughSpace(requestVolumes, pool, plan.getClusterId());
     }
 
+    private boolean managedStoragePoolCanScale(StoragePool storagePool, DeploymentPlan plan) {
+        return storageUtil.managedStoragePoolCanScale(storagePool, plan.getClusterId(), plan.getHostId());
```

Comment:
Why did you need to encapsulate this call into a method?

Reply:
I think it used to do more. I changed this.